subsys/bluetooth/host/hci_core.c

```diff
@@ -3724,6 +3724,9 @@ int bt_disable(void)
 	/* Some functions rely on checking this bitfield */
 	memset(bt_dev.supported_commands, 0x00, sizeof(bt_dev.supported_commands));
 
+    /* If random address was set up - clear it */
+	memset(&bt_dev.random_addr, 0x00, sizeof(bt_dev.random_addr));
```

Comment:
maybe use BT_ADDR_ANY for clearing?
bt_addr_le_copy(&bt_dev.random_addr, BT_ADDR_LE_ANY);

Reply:
good point! already changed